### PR TITLE
Feature/add hub method for displaying question

### PR DESCRIPTION
**Pull request by khlopkov**
I suppose, that hub logic for getting, hiding, and displaying question should be moved to own hub. 
What do you think about that?

### CODE REVIEW

File: Arcadia.Ask/Questions/DisplayedQuestion.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+﻿namespace Arcadia.Ask.Questions
+{
+    using System;
+    using System.Threading;
+
+    public class DisplayedQuestion : IDisplayedQuestion
+    {
+        private Guid currentDisplayedQuestionId;
+        private readonly ReaderWriterLock rwLock = new ReaderWriterLock();
```
Comment (santee) [inline Arcadia.Ask/Questions/DisplayedQuestion.cs]: why do you need ReaderWriterLock? You'd be good with just `public Guid CurrentDisplayedQuestionId { get; set; }`

File: Arcadia.Ask/Hubs/QuestionsHub.cs [hunk 2]
```diff
@@ -111,7 +141,8 @@ private QuestionDto EntityToDto(QuestionEntity entity)
                 Text = entity.Text,
                 PostedAt = entity.PostedAt,
                 IsApproved = entity.IsApproved,
-                Votes = entity.Votes?.Count ?? 0
+                Votes = entity.Votes?.Count ?? 0,
+                IsDisplayed = this.displayedQuestion.CurrentDisplayedQuestionId == entity.QuestionId
```
Comment (santee) [inline Arcadia.Ask/Hubs/QuestionsHub.cs]: funny thing is that you were worried about thread safety of assignment, but it's still possible here that CurrentDisplayedQuestionId changes during the loop :)

Comment (santee) [inline Arcadia.Ask/Hubs/QuestionsHub.cs]: this `displayed`, is it for having a question on a big screen?
1) it's not a property of a Question, why should it care about whether it is displayed on big screen or not? Maybe I want to have multiple screens, do I need to change my question model every time? It should be a separate model
2) and a separate hub - this one should be responsible for questions only, let display status be separated

**Comment by sponkrashin** [comment]
@santee, by the way, why don't we still have issues on board related to current PR and other PRs before?

**Comment by santee** [comment]
@sponkrashin why do think? Because i'm lazy.

**Comment by sponkrashin** [comment]
@santee, don't be lazy, be active! That's what I think :)

**Comment by sponkrashin** [comment]
Also I think, as I've already mentioned, that we need issues with some description to understand what we need to review at all.

File: Arcadia.Ask/Hubs/DisplayQuestionHub.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            try
+            {
+                var displayedQuestionId = this.displayedQuestion.CurrentDisplayedQuestionId;
+                var question = await this.questionsStorage.GetQuestion(displayedQuestionId);
+
+                return new QuestionDto(question);
+            }
+            catch (QuestionNotFoundException)
+            {
+                await this.Clients.All.DisplayedQuestionHidden();
```
Comment (santee) [inline Arcadia.Ask/Hubs/DisplayQuestionHub.cs]: what's your goal behind this call?

File: Arcadia.Ask/Hubs/DisplayQuestionHub.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+        [Authorize(Roles = RoleNames.Moderator)]
+        public async Task HideQuestion()
+        {
+            this.displayedQuestion.CurrentDisplayedQuestionId = Guid.Empty;
+
+            await this.Clients.All.DisplayedQuestionHidden();
+        }
+
+        public async Task<QuestionDto> GetDisplayedQuestion()
```
Comment (santee) [inline Arcadia.Ask/Hubs/DisplayQuestionHub.cs]: why do you need this method? You already have these models on the client side. Just return an id.

File: Arcadia.Ask/Startup.cs [hunk 5]
```diff
@@ -82,6 +84,7 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
             app.UseSignalR(routes =>
             {
                 routes.MapHub<QuestionsHub>("/questions");
+                routes.MapHub<DisplayQuestionHub>("/displayed_question");
```
Comment (santee) [inline Arcadia.Ask/Startup.cs]: just a matter of preference, could you please change _ to -? :)

File: Arcadia.Ask/Questions/Exceptions/QuestionNotApprovedException.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Arcadia.Ask.Questions.Exceptions
+{
+    using System;
+
+    public class QuestionNotApprovedException : Exception
+    {
+        public QuestionNotApprovedException(Guid questionId)
+            : base($"Question with id {questionId} was not approved, yet")
```
Comment (sponkrashin) [inline Arcadia.Ask/Questions/Exceptions/QuestionNotApprovedException.cs]: Excessive comma.

File: Arcadia.Ask/Hubs/DisplayQuestionHub.cs [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+        [Authorize(Roles = RoleNames.Moderator)]
+        public async Task HideQuestion()
+        {
+            this.displayedQuestion.CurrentDisplayedQuestionId = Guid.Empty;
+
+            await this.Clients.All.DisplayedQuestionHidden();
+        }
+
+        public Task<Guid> GetDisplayedQuestionId()
```
Comment (sponkrashin) [inline Arcadia.Ask/Hubs/DisplayQuestionHub.cs]: This method can be called by any user, even unauthorized? Probably, not.

Comment (khlopkov) [reply Arcadia.Ask/Hubs/DisplayQuestionHub.cs]: No. The conception of the app is that displayed question must be shown for every user on big screen, but only moderator can display it or hide.

**Pull request merged**